build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/AndroidCompose.kt

```diff
@@ -52,20 +51,19 @@ internal fun Project.configureAndroidCompose(
     }
 
     extensions.configure<ComposeCompilerGradlePluginExtension> {
-        fun Provider<String>.onlyIfTrue() = flatMap { provider { it.takeIf(String::toBoolean) } }
-        fun Provider<*>.relativeToRootProject(dir: String) = map {
-            isolated.rootProject.projectDirectory
-                .dir("build")
-                .dir(projectDir.toRelativeString(rootDir))
-        }.map { it.dir(dir) }
 
-        project.providers.gradleProperty("enableComposeCompilerMetrics").onlyIfTrue()
-            .relativeToRootProject("compose-metrics")
-            .let(metricsDestination::set)
+        val isComposeCompilerReportsAndMetricsEnabled = providers.gradleProperty("enableComposeCompilerReportsAndMetrics").orNull.toBoolean()
+        if (isComposeCompilerReportsAndMetricsEnabled) {
+            /**
+             * Set path where the `compose-metrics` and `compose-reports`
+             * being written up to the build/***modules-name***/[dir] location.
+             */
+            fun setComposeMetricsAndReportLocation(dir: String) =
```

Comment:
Again, this is not a "setter" so it should not be named like this. I've given multiple examples. Now I'll let the maintainers decide what they prefer here.